static/js/utils/nl_interface_utils.ts

```diff
@@ -28,6 +30,36 @@ const VERSION_PARAM_PREFIX = "&entry.1420739572=";
 const QUERY_CHAIN_PARAM_PREFIX = "&entry.1836374054=";
 const DEBUG_INFO_PARAM_PREFIX = "&entry.1280679042=";
 
+export const CHART_FEEDBACK_SENTIMENT = {
+  THUMBS_DOWN: "THUMBS_DOWN",
+};
+
+interface ChartId {
+  queryIdx: number;
+  categoryIdx: number;
+  blockIdx: number;
+  columnIdx: number;
+  tileIdx: number;
+}
+
+// Given the dom ID of a chart, extract its relevant indexes in
+// the query session and SubjectPageConfig proto.
+export function getNLChartId(idStr: string): ChartId {
```

Comment:
Might need some integration test to make sure the id format does not change

Reply:
Added a TODO and it will be part of webdriver test like we discussed.